docs/changelog.md

```diff
@@ -14,6 +14,9 @@
 ### Breaking Changes
 * The `ContainerCreationConfigBuilder` constructor no longer takes in a test volume mountpoint
 * Added a `ContainerCreationConfigBuilder.WithTestVolumeMountpoint` for specifying the test volume mountpoint, which should be used instead if the default test volume mountpoint of `/kurtosis-test-volume` isn't acceptable
+* Replaced the argument `ContainerConfigFactory` in `AddService`and `AddServiceToPartition`with two arguments `ContainerCreationConfig`and an anonymous function which should returns `ContainerRunConfig`type;
+  users should use `ContainerCreationConfig` struct, and the function that was defined in `GetRunCongig` in the `ContainerConfigFactory` implementations as the new arguments
+* Removed `ContainerConfigFactory` interface
```

Comment:
Make sure to put a remediation, e.g. "Users should instead feed the `ContainerCreationConfig` and `ContianerRunConfig` values directly to `NetworkContext.AddService` or `NetworkContext.AddServiceToPartition`"